service/matching/version_rule_helpers.go

```diff
@@ -291,25 +312,27 @@ func ListWorkerVersioningRules(
 	}, nil
 }
 
-// checkAssignmentConditions returns an error if the new set of assignment rules don't meet the following requirements:
+// checkAssignmentConditions checks for validity conditions that must be assessed by looking at the entire set of rules.
+// It returns an error if the new set of assignment rules don't meet the following requirements:
 // - No more rules than dynamicconfig.VersionAssignmentRuleLimitPerQueue
 // - If `requireUnfiltered`, ensure at least one unfiltered rule still exists
 func checkAssignmentConditions(g *persistencepb.VersioningData, maxARs int, requireUnfiltered bool) error {
-	activeRules := getActiveAssignmentRules(slices.Clone(g.GetAssignmentRules()))
+	activeRules := getActiveAssignmentRules(g.GetAssignmentRules())
 	if cnt := len(activeRules); maxARs > 0 && cnt > maxARs {
 		return serviceerror.NewFailedPrecondition(fmt.Sprintf("update exceeds number of assignment rules permitted in namespace (%v/%v)", cnt, maxARs))
 	}
 	if requireUnfiltered && !containsUnfiltered(activeRules) {
-		return serviceerror.NewFailedPrecondition("update breaks requirement that at least one assignment rule must have no ramp or hint")
+		return serviceerror.NewFailedPrecondition("there must exist at least one 'unfiltered' assignment rule (one with no ramp or hint)")
```

Comment:
since we don't have hint yet
```suggestion
		return serviceerror.NewFailedPrecondition("there must exist at least one fully-ramped assignment rule")
```